### PR TITLE
Veoss 557

### PR DESCRIPTION
**Pull request by oliwiaKot**
contains changes from VEOSS-556 https://github.com/hmrc/one-stop-shop-returns-frontend/pull/137

### CODE REVIEW

File: app/services/SalesAtVatRateService.scala [hunk 1]
```diff
@@ -56,20 +57,6 @@ class SalesAtVatRateService @Inject()() {
     )
   }
 
-  def getTotalNetSales(userAnswers: UserAnswers): BigDecimal   = {
```
Comment (darbyo) [inline app/services/SalesAtVatRateService.scala:59]: I assume these weren't used anywhere else anymore?

**Review by darbyo — APPROVED** [review]

File: test/services/SalesAtVatRateServiceSpec.scala [hunk 2]
```diff
@@ -453,5 +428,97 @@ class SalesAtVatRateServiceSpec extends SpecBase {
...
+        service.getTotalVatOwedAfterCorrections(completeSalesFromNIUserAnswers) mustBe BigDecimal(1000)
+      }
+
+      "must return total when NI doesn't exist and EU does exist" in {
+        val answers = emptyUserAnswers
+          .set(SoldGoodsFromEuPage,true).success.value
+          .set(CountryOfSaleFromEuPage(index), Country("HR", "Croatia")).success.value
+          .set(CountryOfConsumptionFromEuPage(index, index), Country("BE", "Belgium")).success.value
+          .set(VatRatesFromEuPage(index, index), List(twentyPercentVatRate)).success.value
+          .set(NetValueOfSalesFromEuPage(index0, index0, index0), BigDecimal(100)).success.value
```
Comment (darbyo) [inline test/services/SalesAtVatRateServiceSpec.scala:455]: In these tests can we use either all index or all index0. I know theyre the same, but just to make it clear theres no different indexing happening here

File: app/controllers/CheckYourAnswersController.scala [hunk 3]
```diff
@@ -77,18 +79,32 @@ class CheckYourAnswersController @Inject()(
         ).flatten
       ).withCssClass("govuk-!-margin-bottom-9")
 
-      val totalVatToCountries = service.getVatOwedToEuCountries(request.userAnswers)
-      val totalVatOnSales = service.getTotalVatOnSales(request.userAnswers)
+      val containsCorrections = request.userAnswers.get(AllCorrectionPeriodsQuery).isDefined
+
+      val totalVatToCountries = service.getVatOwedToEuCountries(request.userAnswers).filter(vat => vat.totalVat > 0)
+      val noPaymentDueCountries = service.getVatOwedToEuCountries(request.userAnswers).filter(vat => vat.totalVat <= 0)
+      val totalVatOnSales = service.getTotalVatOwedAfterCorrections(request.userAnswers)
```
Comment (darbyo) [inline app/controllers/CheckYourAnswersController.scala:86]: would be good if we could have a look at the failing test and add in something to tell the mock for this service to return what we expect. As there is logic associated with the value returned from here to conditionally show text. Its different to what they were before hand as they defaulted to zero so I dont think they had logic associated with them

File: app/viewmodels/checkAnswers/corrections/CorrectionReturnPeriodSummary.scala [hunk 4]
```diff
@@ -43,9 +44,29 @@ object CorrectionReturnPeriodSummary {
...
               .withVisuallyHiddenText(messages("correctionReturnPeriod.change.hidden"))
           )
         )
     }
+
+  def getAllRows(answers: UserAnswers)(implicit messages: Messages): Option[SummaryListRow] = {
+    val periods = answers.get(AllCorrectionPeriodsQuery).getOrElse(List.empty).map{
+      correction => correction.correctionReturnPeriod.displayText
+    }
+    if(periods.nonEmpty){
```
Comment (darbyo) [inline app/viewmodels/checkAnswers/corrections/CorrectionReturnPeriodSummary.scala:57]: Now we have logic in this method could we add a super quick test for each side of this 'if'

**Review by darbyo — CHANGES_REQUESTED** [review]

File: app/services/SalesAtVatRateService.scala (hunk 1 above)
Comment (oliwiaKot) [reply app/services/SalesAtVatRateService.scala:59]: that's right, they weren't used anymore

File: test/services/SalesAtVatRateServiceSpec.scala (hunk 2 above)
Comment (oliwiaKot) [reply test/services/SalesAtVatRateServiceSpec.scala]: done now

**Review by DevAndyroo — APPROVED** [review]

File: app/viewmodels/checkAnswers/corrections/CorrectionReturnPeriodSummary.scala (hunk 4 above)
Comment (oliwiaKot) [reply app/viewmodels/checkAnswers/corrections/CorrectionReturnPeriodSummary.scala]: got it in the controller specs

File: app/controllers/CheckYourAnswersController.scala (hunk 3 above)
Comment (oliwiaKot) [reply app/controllers/CheckYourAnswersController.scala]: we didn't need to mock the service in the tests for this as we are using appropriate user answers for each one, so we can use the actual service

**Pull request merged**